packages/server/lib/controllers/auth/postBill.ts

```diff
@@ -0,0 +1,209 @@
+        });
+        return;
+    }
+
+    const queryStringVal = queryStringValidation.safeParse(req.query);
+    if (!queryStringVal.success) {
+        res.status(400).send({
+            error: { code: 'invalid_query_params', errors: zodErrorToHTTP(queryStringVal.error) }
+        });
+        return;
+    }
+
+    const paramVal = paramValidation.safeParse(req.params);
+    if (!paramVal.success) {
+        res.status(400).send({
+            error: { code: 'invalid_uri_params', errors: zodErrorToHTTP(paramVal.error) }
+        });
+        return;
+    }
+
+    const { account, environment } = res.locals;
+    const { username: userName, password: password, organization_id: organizationId, dev_key: devkey }: PostPublicBillAuthorization['Body'] = val.data;
+    const { connection_id: receivedConnectionId, params, hmac }: PostPublicBillAuthorization['Querystring'] = queryStringVal.data;
+    const { providerConfigKey }: PostPublicBillAuthorization['Params'] = paramVal.data;
+    const connectionConfig = params ? getConnectionConfig(params) : {};
+
+    let logCtx: LogContext | undefined;
+
+    try {
+        logCtx = await logContextGetter.create(
+            {
+                operation: { type: 'auth', action: 'create_connection' },
+                meta: { authType: 'bill' },
+                expiresAt: defaultOperationExpiration.auth()
+            },
+            { account, environment }
+        );
+        void analytics.track(AnalyticsTypes.PRE_BILL_AUTH, account.id);
+
+        await hmacCheck({
+            environment,
+            logCtx,
+            providerConfigKey,
+            connectionId: receivedConnectionId,
+            hmac,
+            res
+        });
+
+        const config = await configService.getProviderConfig(providerConfigKey, environment.id);
+        if (!config) {
+            await logCtx.error('Unknown provider config');
+            await logCtx.failed();
+            res.status(404).send({ error: { code: 'unknown_provider_config' } });
+            return;
+        }
+
+        const provider = getProvider(config.provider);
+        if (!provider) {
+            await logCtx.error('Unknown provider');
+            await logCtx.failed();
+            res.status(404).send({ error: { code: 'unknown_provider_template' } });
+            return;
+        }
+
+        if (provider.auth_mode !== 'BILL') {
+            await logCtx.error('Provider does not support BILL auth', { provider: config.provider });
+            await logCtx.failed();
+            res.status(400).send({ error: { code: 'invalid_auth_mode' } });
+            return;
+        }
+
+        await logCtx.enrichOperation({ integrationId: config.id!, integrationName: config.unique_key, providerName: config.provider });
+
+        const { success, error, response: credentials } = await connectionService.getBillCredentials(provider, userName, password, organizationId, devkey);
+
+        console.log(success);
+        console.log(credentials);
```

Comment:
console.log 🪓

Reply:
🤦‍♂️, updated  this with the latest [commit](https://github.com/NangoHQ/nango/pull/2852/commits/2aeacff57d36fa34b3461b7995d1291ec5c1dbd8), thanks.